Apsitvarkom.Api/Controllers/PollutedLocationController.cs

```diff
@@ -71,7 +84,41 @@ public async Task<ActionResult<PollutedLocationGetResponse>> GetById(string id)
         try
         {
             var location = await _repository.GetByPropertyAsync(x => x.Id.ToString() == id);
-            return location is not null ? Ok(location) : NotFound($"Polluted location with the specified id '{id}' was not found.");
+
+            if (location is null) return NotFound($"Polluted location with the specified id '{id}' was not found.");
+
+            var mappedLocation = _mapper.Map<PollutedLocationGetResponse>(location);
+            if (mappedLocation is null) return StatusCode(StatusCodes.Status500InternalServerError);
+
+            return Ok(mappedLocation);
+        }
+        catch (Exception)
+        {
+            return StatusCode(StatusCodes.Status500InternalServerError);
+        }
+    }
+
+    [ProducesResponseType(StatusCodes.Status201Created)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(List<string>))]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    [HttpPost("Create")]
+    public async Task<ActionResult<PollutedLocationGetResponse>> Create(PollutedLocationCreateRequest pollutedLocationCreateRequest)
+    {
+        var validationResult = await _pollutedLocationValidator.ValidateAsync(pollutedLocationCreateRequest);
+        if (!validationResult.IsValid) return BadRequest(validationResult.Errors.Select(e => e.ErrorMessage).ToList());
+
+        var mappedPollutedLocation = _mapper.Map<PollutedLocation>(pollutedLocationCreateRequest);
+        if (mappedPollutedLocation is null) return StatusCode(StatusCodes.Status500InternalServerError);
+
+        try
+        {
+            var result = await _repository.InsertAsync(mappedPollutedLocation);
+
+            var response = _mapper.Map<PollutedLocationGetResponse>(result);
+            // If failed to map the response, still return 201 because the location was inserted into the database
+            if (response is null) return StatusCode(StatusCodes.Status201Created);
+
+            return CreatedAtAction(nameof(GetById), new { response.Id }, response);
```

Comment:
This also doesn't seem necessary. We have `PollutedLocationGetResponse` at our hands. Why do we need to use a query to get it again?

Reply:
You're misunderstanding `CreatedAtAction` a bit. It tells the user that the resource they requested was created *and can be retrieved by a certain action*, in our case that action is `GetById` and we also signal that the user would need to call that action with the `response.Id` argument value. This can be seen on swagger:

![image](https://user-images.githubusercontent.com/29711974/202867056-491d9073-8090-42fe-a5e2-75787e64b9af.png)

Reply:
Yeah, interpreted this wrong. My bad.

Also, that's cool. Didn't know about it.